File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/viewmodels/ProfileViewModel.kt
package family.aladdin.android.viewmodels

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

/**
 * 👤 Profile View Model
 * Логика для экрана профиля
 * Источник: iOS ProfileViewModel.swift
 */

class ProfileViewModel : ViewModel() {
    
    // MARK: - State
    
    private val _userName = MutableStateFlow("Сергей Хлыстов")
    val userName: StateFlow<String> = _userName
    
    private val _userEmail = MutableStateFlow("sergey@aladdin.family")
    val userEmail: StateFlow<String> = _userEmail
    
    private val _userPhone = MutableStateFlow("+7 (999) 123-45-67")
    val userPhone: StateFlow<String> = _userPhone
    
    private val _registrationDate = MutableStateFlow("15 сентября 2025")
    val registrationDate: StateFlow<String> = _registrationDate
    
    private val _subscriptionType = MutableStateFlow("Premium")
    val subscriptionType: StateFlow<String> = _subscriptionType
    
    private val _subscriptionEndDate = MutableStateFlow("31.12.2025")
    val subscriptionEndDate: StateFlow<String> = _subscriptionEndDate
    
    private val _threatsBlocked = MutableStateFlow(47)
    val threatsBlocked: StateFlow<Int> = _threatsBlocked
    
    private val _familyMembers = MutableStateFlow(4)
    val familyMembers: StateFlow<Int> = _familyMembers
    
    private val _devices = MutableStateFlow(8)
    val devices: StateFlow<Int> = _devices
    
    // MARK: - Public Methods
    
    /**
     * Редактировать профиль
     */
    fun editProfile() {
        println("Show edit profile sheet")
        // В production: открыть форму редактирования
    }
    
    /**
     * Изменить пароль
     */
    fun changePassword() {
        println("Show change password screen")
        // В production: форма смены пароля
    }
    
    /**
     * Управление 2FA
     */
    fun manageTwoFactor() {
        println("Manage 2FA settings")
        // В production: настройки двухфакторной аутентификации
    }
    
    /**
     * Показать активные сеансы
     */
    fun showActiveSessions() {
        println("Show active sessions list")
        // В production: список активных сессий
    }
    
    /**
     * Удалить аккаунт
     */
    fun deleteAccount() {
        println("Show delete account confirmation")
        // В production: подтверждение удаления + API вызов
    }
    
    /**
     * Загрузить данные профиля
     */
    fun loadProfileData() {
        // В production: API запрос на получение данных пользователя
        println("Loading profile data from API...")
    }
}

File: BACKUP_SCREENS_20251028_200516/ProfileViewModel.swift
import SwiftUI

/// 👤 Profile View Model
/// Логика для экрана профиля
class ProfileViewModel: ObservableObject {
    
    @Published var userName: String = "Сергей Хлыстов"
    @Published var userEmail: String = "sergey@aladdin.family"
    @Published var userPhone: String = "+7 (999) 123-45-67"
    @Published var registrationDate: String = "15 сентября 2025"
    @Published var subscriptionType: String = "Premium"
    @Published var subscriptionEndDate: String = "31.12.2025"
    @Published var threatsBlocked: Int = 47
    @Published var familyMembers: Int = 4
    @Published var devices: Int = 8
    
    func editProfile() {
        print("Show edit profile sheet")
    }
    
    func changePassword() {
        print("Show change password")
    }
    
    func manageTwoFactor() {
        print("Manage 2FA")
    }
    
    func showActiveSessions() {
        print("Show active sessions")
    }
    
    func deleteAccount() {
        print("Show delete account confirmation")
    }
}



